### ISSUE TITLE
Lost Custom Bookmarks

### ISSUE THREAD

**Issue by gcbartlett** [leggett/simplify#829]
After restarting my computer today, all my saved (custom) bookmarks (searches) were lost.  Unlike in https://github.com/leggett/simplify/issues/817, restarting Simplify v3.1.21 or the browser did not bring them back.

Additionally, closing the browser and restarting it also loses newly created searches.

I am using Microsoft Edge (129.0.2792.79) browser.

**Comment by gcbartlett** [comment]
System Report (after lost bookmarks):

Simplify v.3.1.21 u.0 configuration: aAX appNavOff autoAdvance avatarsZoom bundleInbox bundleLists caA caE caI caL cfC cfF cfO cfR cfU composeActions composeFab composeHeightMd dateGroup fontList fontMsg fontNav gaArchive gaRemove geminiMin genSumMin groupHideFirst groupInbox groupOtherLists hideAds hideEmptySections hideExtWarn hideFileIcons hideImportance hideInboxHideAllLink hideLabelChips hideListCount hideMsgCount hideMsgReactions hideSignatures hideTabIcons highDensity imgTheme inInbox inList invertCompose izBg23 listWidthFull matYou mediumTheme msgWidthFull navOpen newNav newUI reminders rightImportance rightLabels rightStar sendLater shortLabels simplify simplifyPrefsOpen- 

System: Mozilla/5.0 (Windows NT 10.0 Win64 x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36 - gmail.pinto-server_20241003.04_p0 - Window: 1411 x 1407 (92pct zoom) - Language: en

**Comment by leggett** [comment]
I'm so sorry that happened Gary!! 

In my effort to make things as private and secure as possible, I may have swung too far by storing all Simplify preferences and things like bookmarks in local storage with your browser. I'm finding that some users have their browser set to(or the browsers themselves default to) erase this local storage. I haven't personally experienced this but I know it is possible. 

I am working on a big update where I periodically sync the preferences to my servers. I'll let users turn this off if they wish and download their preferences locally. 

I wish there was more I could do or say but I fear they may be gone. :(

**If you want to be absolutely sure, you can run these commands in the developer console:**

Open a tab with Gmail in your main browser profile.

Open the console under Tools → Developer → JavaScript Console
<img width="504" alt="image" src="https://github.com/user-attachments/assets/c738e1ee-c74f-47cf-93e0-f34173aada0b">

Type in `localStorage.savedSearches` and see if anything is found:
<img width="318" alt="image" src="https://github.com/user-attachments/assets/b993ccd2-d015-4237-8a82-000f5a44aa5f">

If you do see anything, we can recover the bookmarks. If not, I'm afraid they are lost.

**Comment by gcbartlett** [comment]
Thank you for the quick response, Michael.

I can confirm that my local storage is empty:
[image: image.png]
I am happy to wait for the next update (my searches are easy enough to
recreate).

Do you have any idea which settings for Edge (or Windows) control whether
(and where?) the local storage is persisted?

Please feel free to let me know if I can help debug this any further or try
other workarounds.

Thanks again,
Gary

**Comment by gcbartlett** [comment]
I found an Edge setting that included a "*Clear on exit*" URL pattern
for *https://[*.]google.com:443
<http://google.com:443>* (it might've not been exactly that, as I deleted
it before grabbing it).  Deleting this URL causes the Simplify local
storage settings to be preserved across browser restarts.

[image: image.png]
I don't know what added these URL patterns in the first place, though.

I also think that saving the settings outside the browser would be
preferable because clearing caches is something I might do anyway from time
to time.

Hope this workaround helps.

Regards,
Gary
